client/opkclient_test.go

```diff
@@ -104,43 +105,30 @@ func TestClient(t *testing.T) {
 			}
 			jktstr := string(data)
 
-			pubkey, err := op.PublicKey(context.Background(), nil)
-			require.NoError(t, err, tc.name)
+			providerVerifier := verifier.NewProviderVerifier(provider.Issuer(), provider.CommitmentClaim(), verifier.ProviderVerifierOpts{})
 
-			pub, err := jwk.FromRaw(pubkey)
+			pubkey, err := providerVerifier.ProviderPublicKey(context.Background(), pkt)
+			if err != nil {
+				t.Fatal(err)
+			}
 			require.NoError(t, err, tc.name)
 
-			thumbprint, err := pub.Thumbprint(crypto.SHA256)
+			thumbprint, err := pubkey.Thumbprint(crypto.SHA256)
 			require.NoError(t, err, tc.name)
 
 			thumbprintStr := string(util.Base64EncodeForJWT(thumbprint))
 			require.Equal(t, jktstr, thumbprintStr, "jkt header does not match op thumbprint in "+tc.name)
 
-			alg, ok := pkt.ProviderAlgorithm()
+			providerKeyAlgorithm, ok := pkt.ProviderAlgorithm()
 			if !ok {
 				t.Fatal(fmt.Errorf("missing algorithm"))
 			}
 
 			if tc.gq {
-				require.Equal(t, gq.GQ256, alg, tc.name)
-
-				// Verify our GQ signature
-				idt, err := pkt.Compact(pkt.Op)
-				require.NoError(t, err, tc.name)
-
-				opPubKey, err := op.PublicKey(context.Background(), nil)
-				require.NoError(t, err, tc.name)
-
-				sv, err := gq.NewSignerVerifier(opPubKey.(*rsa.PublicKey), client.GQSecurityParameter)
-				require.NoError(t, err, tc.name)
-
-				ok := sv.VerifyJWT(idt)
-				if !ok {
-					t.Fatal(fmt.Errorf("error verifying OP GQ signature on PK Token (ID Token invalid)"))
-				}
+				require.Equal(t, gq.GQ256, providerKeyAlgorithm, tc.name)
```

Comment:
What is your motivation behind removing the test that client produced a valid GQ signature?

Reply:
I think it was to reduce test redundancy particularly when we don't do the same for other algorithms. This should be tested (which it is) in the verifier package. The client tester should focus on testing the inputs and outputs of the client functions which is why it makes sense to check if the pk token has a certain algorithm. 

I also think we've encountered an issue in previous tests where the code used to test wasn't accurate or was assuming too much and this helps to keep everything tight and reliable.

Reply:
If we don't do the same for the other algorithm then we should add a test for the other algorithms rather than remove the test.